### PR TITLE
fix: Adjust Page Size and Remove Vertical Scroll

### LINKED ISSUE #17857: [Bug]: Table shows a scrollbar when trying to fit all items in table window
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When i try to adjust the height of table to make sure no more rows are present, the table still ends up showing an extra row all the time.

### Steps To Reproduce

https://www.loom.com/share/6141092d38ef4be09ce24ed85fd4d8f6

### Public Sample App

_No response_

### Version

Release

### PR DESCRIPTION
**Pull request by gurel**
TL;DR;
The PR includes some improvements to pagination on Table Widget. There are two main changes
1- Adjust the page size so it doesn't need to use a vertical scroll bar for just the last item.
2- When the user focuses on the page size the whole page number is selected so it can easily be set to the new value.

## Description
The table widget for some reason includes a vertical scroll bar always. This interrupts the user experience as it creates another vertical scroll when it's not necessary. Multiple scroll makes it harder for the user to navigate through the window. The table widget already has a fixed pageSize and there is no way to change the pageSize. It is fully calculated by the height of the table. In this PR I've corrected the calculation and removed the 30% check of the last row that was added. 

Secondly changing the page number required a few more clicks that I expected. The user had to click, delete and then type the new value. I've added a selection on blur so they can just click and type.

Media
lMRaObjy8O
https://recordit.co/lMRaObjy8O

Fixes #17857

## Type of change
- Bug fix
- New feature

## How Has This Been Tested?
- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by mohanarpit** [comment]
@gurel Thanks for opening this PR & making Appsmith a better project for everybody. Apologies that we didn't pick this up sooner for review. We'll try to review this code & merge it in as soon as we can.

**Comment by sbalaji1192** [comment]
@mohanarpit The DP link posted by vercel bot throws 404. Has it been deleted because the pr is stale ? Can you trigger a vercel build again?

**Comment by gurel** [comment]
@dilippitchika Yeah, i didn't realize it went away when the app is relased. I'll look into it for sure.

**Comment by dilippitchika** [comment]
Fixes https://github.com/appsmithorg/appsmith/issues/17857
